v2/slogger/rolling_file_appender/state.go

```diff
@@ -0,0 +1,64 @@
+package rolling_file_appender
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+type state struct {
+	LogStartTime time.Time
+}
+
+func newState(logStartTime time.Time) *state {
+	return &state{logStartTime}
+}
+
+func readState(path string) (*state, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, OpenError{path, err}
+	}
+	defer file.Close()
+
+	var decodedState *state
+	decoder := json.NewDecoder(file)
+	if err = decoder.Decode(&decodedState); err != nil {
+		return nil, DecodeError{path, err}
+	}
+
+	return decodedState, nil
+}
+
+func stateExists(path string) (bool, error) {
```

Comment:
It is intended to be `stateExists`.  It just so happens that we use `os.Stat()` to determine whether the state file exists.

The state currently consists only of the creation time of the log file.